packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -70,31 +72,40 @@ export const SelectControl = ( {
 	onInputChange = () => null,
 	onRemove = () => null,
 	onSelect = () => null,
+	clearSearchOnSelect = false,
+	keepMenuOpenOnSelect = false,
 	placeholder,
 	selected,
 }: SelectControlProps ) => {
 	const [ isFocused, setIsFocused ] = useState( false );
 	const [ inputValue, setInputValue ] = useState( '' );
-	const { getSelectedItemProps, getDropdownProps } = useMultipleSelection();
+	const multipleSelection = useMultipleSelection();
 	let selectedItems = selected === null ? [] : selected;
 	selectedItems = Array.isArray( selectedItems )
 		? selectedItems
 		: [ selectedItems ].filter( Boolean );
 	const filteredItems = getFilteredItems( items, inputValue, selectedItems );
 
+	let currentSelectedItem = null;
+	if ( ! multiple && selectedItems.length > 0 ) {
+		currentSelectedItem = selectedItems[ 0 ];
```

Comment:
Heads up that I have multiple changes to how state selection is handled landing in https://github.com/woocommerce/woocommerce/pull/34510 related to accessibility.